ext/session/session.c

```diff
@@ -1600,6 +1618,15 @@ PHPAPI zend_result php_session_start(void) /* {{{ */
 		}
 		return FAILURE;
 	}
+
+	/* Should these be set here, or in session_initialize? */
+	php_session_cleanup_filename();
+	zend_string *session_started_filename = zend_get_executed_filename_ex();
+	if (session_started_filename != NULL) {
+		PS(session_started_filename) = zend_string_copy(session_started_filename);
+		PS(session_started_lineno) = zend_get_executed_lineno();
+	}
```

Comment:
Does this still work if you move this to initialize? As I would agree this seems to make more sense in session_initialize